### PR TITLE
Relax device-spec observables validation

### PR DESCRIPTION
**Pull request by grwlf**
**Context:** Quantum device config now contains only native observables (it means, handled by C-API), while the `device.observables` may contain additional non-native observables supported by the device Python API.

**Description of the Change:** By this PR we relax the observables validation.

**Benefits:** Fix support toml schema 2 configs.

**Possible Drawbacks:** 

**Related GitHub Issues:**
* https://github.com/PennyLaneAI/pennylane-lightning/pull/659
* https://github.com/PennyLaneAI/pennylane-lightning/pull/642

### CODE REVIEW

**Review by erick-xanadu — CHANGES_REQUESTED** [review]
Do you know if this interferes with lightning latest changes? I am thinking specifically about the migration to the new device API.

**Comment by grwlf** [comment]
> Do you know if this interferes with lightning latest changes? I am thinking specifically about the migration to the new device API.

I don't know for sure, but most probably it does not. It is only related to how we interpret the information contained in the toml files.

**Comment by erick-xanadu** [comment]
> I don't know for sure, I think that it does not. It is only related to how we interpret the information contained in the toml files.

Thanks Sergei! I ran the latest/latest action on this branch and it is no different than main/latest/latest. I agree that this might be independent of the new device API.

File: frontend/catalyst/utils/runtime.py [hunk 1]
```diff
@@ -266,11 +236,25 @@ def validate_config_with_device(device: qml.QubitDevice, config: TOMLDocument) -
...
+        spec_gates = filter_out_adjoint(set.union(native, matrix, decomposable))
+        if device_gates != spec_gates:
+            raise CompileError(
+                "Gates in qml.device.operations and specification file do not match.\n"
+                f"Gates that present only in the device: {device_gates - spec_gates}\n"
+                f"Gates that present only in spec: {spec_gates - device_gates}\n"
+            )
+
+        # For observables, we do not have `non-native` section in the config, so we check that
+        # device data superseeds the specification.
```
Comment (erick-xanadu) [inline frontend/catalyst/utils/runtime.py:250]:
```suggestion
        # device data supercedes the specification.
```

**Review by erick-xanadu — APPROVED** [review]

**Pull request merged**